### PR TITLE
Updating Character-Details Rarity adjust available star choices

### PR DESCRIPTION
**Pull request by NickReinlein**
## Context
As referenced in issue https://github.com/svehera/tacticusplanner/issues/157#issue-2123789931, the stars available in the drop-down menu are "impossible". For example, when the rarity "Uncommon" is set, the number of stars available should be 2, 3, or 4 yellow stars. Anything higher would not be possible at this rarity.

## Solution
The stars available is updated when rarity is changed. A new callback function is introduced to allow for this modification.
The stars are now updated based on the newly-selected rarity value. 

This was tested locally with various changes to rarity and stars in the drop-down menus.

## Example
A "Common" with 1 yellow star has their rarity changed to "Epic", the new options are 1, 2 or 3 red stars . These are the only possible options. ~If a unit is upgraded, defined as an increase in their rarity, then the new minimum stars is selected. If a unit is downgraded, then the maximum stars is selected.~ When rarity is changed, the first available entry (minimum stars) is selected to avoid index issues, since the array representing star entries has now changed (and therefore its indexes)

## Video for local test
Please forgive the odd cropping and resolution; I'm working to resolve it 
https://github.com/svehera/tacticusplanner/assets/5471711/ed330ff9-4313-4b87-9703-363b4d0eb77e

### CODE REVIEW

File: src/mobile-routes/characters/character-details.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ﻿import React, { useMemo, useState } from 'react';
-import { ICharacter2, IMaterialRecipeIngredientFull } from '../../models/interfaces';
+import { ICharacter2, IMaterialRecipeIngredientFull } from 'src/models/interfaces';
```
Comment (NickReinlein) [inline src/mobile-routes/characters/character-details.tsx:2]: This was an IDE suggestion. It worked when tested locally. I can revert it if desired

**Comment by NickReinlein** [comment]
Hello :wave: 
This is my first PR to the project so please let me know if anything needs to be adjusted :smiley:

File: src/mobile-routes/characters/character-details.tsx (hunk 1 above)
Comment (svehera) [reply src/mobile-routes/characters/character-details.tsx:2]: it's fine

**Comment by NickReinlein** [comment]
Maybe I missed it. I'll take a look and update

**Comment by NickReinlein** [comment]
I made changes to the approach and re-tested. It seems to work correctly when I try it locally. The only change in functionality is that it now takes the first available entry in the array of new stars, rather than having a logic that took minimum stars when upgrading and maximum when downgrading

The GIF is a bit long but includes several changes to rarity, stars, unlocking and saving. Please take another look when you get the chance.
QxKKlnULea

**Comment by svehera** [comment]
> I made changes to the approach and re-tested. It seems to work correctly when I try it locally. The only change in functionality is that it now takes the first available entry in the array of new stars, rather than having a logic that took minimum stars when upgrading and maximum when downgrading
> 
> The GIF is a bit long but includes several changes to rarity, stars, unlocking and saving. Please take another look when you get the chance.
> 
> QxKKlnULea [ QxKKlnULea ](https://private-user-images.githubusercontent.com/5471711/320298115-b4733c53-115f-43c9-9146-bc80ae2b2e49.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1OTQ2NzgsIm5iZiI6MTcxMjU5NDM3OCwicGF0aCI6Ii81NDcxNzExLzMyMDI5ODExNS1iNDczM2M1My0xMTVmLTQzYzktOTE0Ni1iYzgwYWUyYjJlNDkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDhUMTYzOTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWVkYWVmZTFmMmUzMjkwNzIxMWU4MDAzMzUzZmJlN2Q2YTEzNWMwMTBjYTNlMzcyNWRlYzU3OTgwMTE0ZmU4MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.f8DRtEew45SX7SatGIsTi-RO83S3Jby-ev6GqpMLnNo) [ ](https://private-user-images.githubusercontent.com/5471711/320298115-b4733c53-115f-43c9-9146-bc80ae2b2e49.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1OTQ2NzgsIm5iZiI6MTcxMjU5NDM3OCwicGF0aCI6Ii81NDcxNzExLzMyMDI5ODExNS1iNDczM2M1My0xMTVmLTQzYzktOTE0Ni1iYzgwYWUyYjJlNDkuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDhUMTYzOTM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWVkYWVmZTFmMmUzMjkwNzIxMWU4MDAzMzUzZmJlN2Q2YTEzNWMwMTBjYTNlMzcyNWRlYzU3OTgwMTE0ZmU4MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.f8DRtEew45SX7SatGIsTi-RO83S3Jby-ev6GqpMLnNo)

Sweet! looks good to me!

**Review by svehera — APPROVED** [review]

**Pull request merged**